### PR TITLE
Add deposit service and related functionalities

### PR DESCRIPTION
**Pull request by signature18632**

### CODE REVIEW

File: internal/deposit_service/deposit_relayer_service.go [hunk 1]
```diff
@@ -0,0 +1,173 @@
...
+	"github.com/ethereum/go-ethereum/ethclient"
+)
+
+const depositThreshold uint64 = 128
+const duration = 1 * time.Hour
+
+type DepositIndices struct {
+	LastSeenDepositIndex      uint64
+	LastProcessedDepositIndex uint64
+}
```
Comment (denis468629) [inline internal/deposit_service/deposit_relayer_service.go:24]:
```suggestion
type DepositIndices struct {
	LastSeenDepositIndex      *uint64
	LastProcessedDepositIndex *uint64
}
```

File: internal/deposit_service/deposit_relayer_service.go [hunk 2]
```diff
@@ -0,0 +1,173 @@
...
+
+	lastProcessedDepositResult := <-lastProcessedDepositIndexCh
+	if lastProcessedDepositResult.err != nil {
+		return DepositIndices{}, lastProcessedDepositResult.err
+	}
+
+	return DepositIndices{
+		LastSeenDepositIndex:      lastSeenDepositResult.index,
+		LastProcessedDepositIndex: lastProcessedDepositResult.index,
+	}, nil
```
Comment (denis468629) [inline internal/deposit_service/deposit_relayer_service.go:78]:
```suggestion
	var di DepositIndices
	for {
		if di.LastSeenDepositIndex != nil && di.LastProcessedDepositIndex != nil {
			return di, nil
		}
		select {
		case lastSeenDepositResult := <-lastSeenDepositIndexCh:
			if lastSeenDepositResult.err != nil {
				return DepositIndices{}, lastSeenDepositResult.err
			}
			di.LastSeenDepositIndex = &lastSeenDepositResult.index
		case lastProcessedDepositResult := <-lastProcessedDepositIndexCh:
			if lastProcessedDepositResult.err != nil {
				return DepositIndices{}, lastProcessedDepositResult.err
			}
			di.LastProcessedDepositIndex = &lastProcessedDepositResult.index
		}
	}
```

File: internal/deposit_service/deposit_relayer_service.go [hunk 3]
```diff
@@ -0,0 +1,173 @@
...
+	if err != nil {
+		log.Fatalf("Failed to instantiate a Liquidity contract: %v", err.Error())
+	}
+
+	indices, err := fetchDepositIndices(ctx, liquidity)
+	if err != nil {
+		log.Fatalf("Failed to fetch deposit indices: %v", err.Error())
+	}
+
+	shouldSubmit, err := shouldProcessDeposits(client, liquidity, indices.LastSeenDepositIndex, indices.LastProcessedDepositIndex)
```
Comment (denis468629) [inline internal/deposit_service/deposit_relayer_service.go:151]:
```suggestion
	shouldSubmit, err := shouldProcessDeposits(client, liquidity, *indices.LastSeenDepositIndex, *indices.LastProcessedDepositIndex)
```

File: internal/deposit_service/deposit_relayer_service.go [hunk 4]
```diff
@@ -0,0 +1,173 @@
...
+	if err != nil {
+		log.Fatalf("Error in threshold and time diff check: %v", err)
+		return
+	}
+
+	if !shouldSubmit {
+		return
+	}
+
+	receipt, err := submitDepositRoot(ctx, cfg, client, liquidity, indices.LastSeenDepositIndex)
```
Comment (denis468629) [inline internal/deposit_service/deposit_relayer_service.go:161]:
```suggestion
	receipt, err := submitDepositRoot(ctx, cfg, client, liquidity, *indices.LastSeenDepositIndex)
```

File: internal/deposit_service/deposit_relayer_service.go [hunk 5]
```diff
@@ -0,0 +1,173 @@
...
+	if err != nil {
+		log.Fatalf("Failed to submit deposit root: %v", err.Error())
+	}
+
+	if receipt.Status == types.ReceiptStatusSuccessful {
+		log.Infof("Successfully submitted deposit root")
+	} else {
+		log.Infof("Failed to submit deposit root")
+	}
```
Comment (denis468629) [inline internal/deposit_service/deposit_relayer_service.go:170]:
```suggestion
        if receipt == nil {
          return
        }

	if receipt.Status == types.ReceiptStatusSuccessful {
		log.Infof("Successfully submitted deposit root")
	} else {
		log.Infof("Failed to submit deposit root")
	}
```

**Comment by signature18632** [comment]
Thank you for the reviewing, I fixed the code based on feedback.

**Pull request merged**